perf(ivy): avoid repeat global state accesses in i18n instructions

crisbeto opened the pull request:
Removes repeat global state accesses from the i18n instructions in the cases where we have the information available already.

petebacondarwin commented on packages/core/src/render3/i18n.ts:
```diff
@@ -656,36 +656,36 @@ export function ɵɵi18nPostprocess(
+function i18nEndFirstPass(lView: LView, tView: TView) {
   ngDevMode && assertEqual(
-                   viewData[BINDING_INDEX], viewData[TVIEW].bindingStartIndex,
+                   lView[BINDING_INDEX], lView[TVIEW].bindingStartIndex,
```
Can this be:

```suggestion
                   lView[BINDING_INDEX], tView.bindingStartIndex,
```

petebacondarwin commented on packages/core/src/render3/i18n.ts:
```diff
@@ -711,9 +711,8 @@ function createDynamicNodeAtIndex(
-    index: number, createOpCodes: I18nMutateOpCodes, icus: TIcu[] | null,
-    viewData: LView): number[] {
-  const renderer = getLView()[RENDERER];
+    index: number, createOpCodes: I18nMutateOpCodes, viewData: LView): number[] {
```
Rename `viewData` to `lView` for consistency?

ocombe commented on packages/core/src/render3/i18n.ts:
```diff
@@ -711,9 +711,8 @@ function createDynamicNodeAtIndex(
-    viewData: LView): number[] {
-  const renderer = getLView()[RENDERER];
+    index: number, createOpCodes: I18nMutateOpCodes, viewData: LView): number[] {
+  const renderer = viewData[RENDERER];
```
not sure if that's exactly the same, depending on the nesting?

crisbeto replied:
From what I could tell it's the same. I also tried adding a `console.log(viewData === getLView())` and it came back true for all of the i18n acceptance tests.

crisbeto committed:
perf(ivy): avoid repeat global state accesses in i18n instructions

Removes repeat global state accesses from the i18n instructions in the cases where we have the information available already.

crisbeto commented:
The feedback has been addressed @petebacondarwin.

petebacondarwin reviewed (approved):
I don't have enough understanding of the innards of ivy to know if caching the `LVIEW` rather than calling `getLView()` could break. But otherwise LGTM

pkozlowski-opensource reviewed (approved):
LGTM

AndrewKushnir commented:
- [VE Presubmit](https://test.corp.google.com/ui#id=OCL:272085278:BASE:272085297:1569884273674:ebb44ccd)
- [Ivy Presubmit](http://test/OCL:272085278:BASE:272085544:1569884350750:1bae4511)

AndrewKushnir commented:
FYI, VE and Ivy presubmits are successful. Thank you.
